sdk/python/kfp/v2/components/component_factory.py

```diff
@@ -60,11 +60,11 @@ def _python_function_name_to_component_name(name):
 
 _INSTALL_PYTHON_PACKAGES_SCRIPT = '''
 if ! [ -x "$(command -v pip)" ]; then
-    python3 -m ensurepip || python3 -m ensurepip --user || apt-get install python3-pip
+    python3 -m ensurepip 2>&1 || python3 -m ensurepip --user 2>&1 || apt-get install python3-pip 2>&1
 fi
 
 PIP_DISABLE_PIP_VERSION_CHECK=1 python3 -m pip install --quiet \
-    --no-warn-script-location {package_list} && "$0" "$@"
+    --no-warn-script-location {package_list} 2>&1 && "$0" "$@"
```

Comment:
The lines other than this file is just golden snapshot file for test.

Reply:
This could also "hide" real errors that may be emitted during installing user specified packages.
Trying to understand the user impact, would it be possible for your system to filter and ignore the specific warning/error on running pip as root?

Reply:
@chensun Ah, I couldn't find that here user can specify their packages. I can understand your point of view and agree with you. So, will handle by ourselves and close this issue.
Thank you so much for your review.